ibis/sql/compiler.py

```diff
@@ -753,15 +760,22 @@ def _is_subquery(self, node):
 
         return False
 
-    def _visit_table(self, expr, in_subquery=False):
+    def _visit_table(self, expr, in_subquery=False, visit_cache=None, visit_table_cache=None):
+        if visit_table_cache is None:
+            visit_table_cache = set()
+
+        if (id(expr), in_subquery) in visit_table_cache:
+            return
+        visit_table_cache.add((id(expr), in_subquery))
+
         node = expr.op()
 
         if isinstance(node, (ops.PhysicalTable, ops.SelfReference)):
             self._ref_check(node, in_subquery=in_subquery)
 
         for arg in node.flat_args():
             if isinstance(arg, ir.Expr):
-                self._visit(arg, in_subquery=in_subquery)
+                self._visit(arg, in_subquery=in_subquery, visit_cache=visit_cache, visit_table_cache=visit_table_cache)
```

Comment:
Can you rebase and also run `flake8` on the codebase to fix the code style (80-char line length, mostly)? Thank you!